ddtrace/internal/telemetry/writer.py

```diff
@@ -165,19 +156,25 @@ def _stop_service(self, *args, **kwargs):
     def add_event(self, payload, payload_type):
         # type: (Dict, str) -> None
         """
-        Adds a Telemetry Request to the TelemetryWriter request buffer
+        Adds a Telemetry event to the TelemetryWriter event buffer
 
-        :param Dict payload: stores a formatted telemetry request
+        :param Dict payload: stores a formatted telemetry event
         :param str payload_type: The payload_type denotes the type of telmetery request.
             Payload types accepted by telemetry/proxy v1: app-started, app-closing, app-integrations-change
         """
-        with self._lock:
-            if not self._enabled:
-                return
-
-            request = self._create_telemetry_request(payload, payload_type, self._sequence)
-            self._sequence += 1
-            self._events_queue.append(request)
+        if self._enabled:
+            event = {
+                "tracer_time": int(time.time()),
+                "runtime_id": get_runtime_id(),
+                "api_version": "v1",
+                "seq_id": next(self._sequence),
+                "debug": str(self._debug).lower(),
+                "application": get_application(config.service, config.version, config.env),
+                "host": get_host_info(),
+                "payload": payload,
+                "request_type": payload_type,
+            }
+            self._events_queue.append(event)
```

Comment:
Does it matter the event's position on the `_events_queue`? It seems to me that even if the `seq_id` is obtained and incremented atomically this doesn't necessarily ensure that the event is placed onto the queue in the exact same position as it would have previous to this change.

Reply:
Nope the position in the events queue does not matter. The only thing that matters is the timestamp and sequence ids in payloads. Also the sequence id is not used by the backend it is only used system tests.